checkers/rules/rules.go

```diff
@@ -808,3 +808,17 @@ func sloppyTestFuncName(m dsl.Matcher) {
 		Where(m["bench"].Text.Matches("^bench(mark)?.*")).
 		Report("function $bench looks like a benchmark helper, consider to change 1st param to 'tb testing.TB'")
 }
+
+//doc:summary Detects bad usage of sync.OnceFunc
+//doc:tags    diagnostic experimental
+//doc:before  sync.OnceFunc(foo)()
+//doc:after   fooOnce := sync.OnceFunc(foo); ...; fooOnce()
+func badSyncOnceFunc(m dsl.Matcher) {
+	m.Match(`$*_; sync.OnceFunc($x); $*_;`).
+		Suggest("sync.OnceFunc($x)()").
```

Comment:
Here I'm proposing to just invoke the result and after that match below will say "hey, it's still wrong".